### PR TITLE
[#47063] Differentiate Nextcloud and OpenProject error responses in API

### PR DESCRIPTION
**Pull request by Kharonus**
- https://community.openproject.org/work_packages/47063

### CODE REVIEW

**Comment by Kharonus** [comment]
@apfohl If you look at the test cases, should we have different reaction in the code to "query creation fails" and "query execution fails"? Right now, query creation cannot fire a `:not_found`, so the test case is academical. But still, what do you think?

**Comment by apfohl** [comment]
> @apfohl If you look at the test cases, should we have different reaction in the code to "query creation fails" and "query execution fails"? Right now, query creation cannot fire a `:not_found`, so the test case is academical. But still, what do you think?

I'm not sure about this. I can imaging a query construction that might need to make HTTP requests. But we would encapsulate these error nicely. I don't like the variant with using symbols for that. Let's see in a future refactoring or so. For now I agree.

File: docs/api/apiv3/paths/storage_files_prepare_upload.yml [hunk 1]
```diff
@@ -64,8 +64,17 @@ post:
             errorIdentifier: urn:openproject-org:api:v3:errors:NotFound
             message: The requested resource could not be found.
       description: |-
-        Returned in either of those cases:
-        - if the storage does not exist or the client does not have sufficient permissions to see it
+        Returned the storage does not exist or the client does not have sufficient permissions to see it.
```
Comment (apfohl) [inline docs/api/apiv3/paths/storage_files_prepare_upload.yml]: That sentence is somehow looking weird. Please have a look again. It doesn't tell me...

File: docs/api/apiv3/paths/storage_files_prepare_upload.yml [hunk 2]
```diff
@@ -64,8 +64,17 @@ post:
...
-        - if the document root file identification does not exist on the storage
+        **Required permission:** view file links
+    '500':
+      content:
+        application/hal+json:
+          schema:
+            $ref: '../components/schemas/error_response.yml'
+          example:
+            _type: Error
+            errorIdentifier: urn:openproject-org:api:v3:errors:OutboundRequestFailure
```
Comment (apfohl) [inline docs/api/apiv3/paths/storage_files_prepare_upload.yml]: What about `OutboundRequest:NotFound`. Now it looks more like a general (🫡) failure. IMO we can be specific here.

File: docs/api/apiv3/paths/storage_files_prepare_upload.yml (hunk 1 above)
Comment (apfohl) [reply docs/api/apiv3/paths/storage_files_prepare_upload.yml]: Maybe an 'if' is missing.

**Review by apfohl — CHANGES_REQUESTED** [review]
These are just suggestions, but please have a look.

File: docs/api/apiv3/paths/storage_files_prepare_upload.yml (hunk 1 above)
Comment (Kharonus) [reply docs/api/apiv3/paths/storage_files_prepare_upload.yml]: added `if`

File: docs/api/apiv3/paths/storage_files_prepare_upload.yml (hunk 2 above)
Comment (Kharonus) [reply docs/api/apiv3/paths/storage_files_prepare_upload.yml]: That's fine for me.

**Review by apfohl — APPROVED** [review]

**Pull request merged**